src/main/java/seedu/duke/component/gates/AndGate.java

```diff
@@ -0,0 +1,21 @@
+package seedu.duke.component.gates;
+
+public class AndGate extends TwoInputGate {
+    private static final String AND = "--(&|--";
+
+    public AndGate(int input1, int input2) {
+        input = input1;
+        secondInput = input2;
+    }
+
+    @Override
+    public int getOutput() {
+        output = input & secondInput;
+        return output;
```

Comment:
Probably clear enough to return immediately.
```suggestion
        return input & secondInput;
```

Reply:
Noted. So I add set output as a separate fn right?

Reply:
I don't think we need that. Reason being we are not guaranteed the inputs are set when building the circuit. Might have to calculate every time, unless `BooleanTemplate` checks for if a new value is set or not.

Any `Gate` would also need a default overloaded constructor in that case.

Reply:
Hmm, makes sense